### ISSUE TITLE
Add VATT model

### ISSUE THREAD

**Issue by johko** [huggingface/transformers#19865]
### Model description

Hey,
as discussed with @NielsRogge a few weeks back, I'd like to work on adding the "VATT: Transformers for Multimodal
Self-Supervised Learning from Raw Video, Audio and Text" model from Google.

It is basically three transformers(Video/Audio/Text) that are trained jointly in an unsupervised manner using contrastive loss functions. For downstreams tasks they fine-tune the Transformers separately, but also explore a version that shares the weights for all modalities.

For Pre-Traning they use text-video-audio triplets from HowTo100M and video-audio pairs from AudioSet. The authors describe how to fine-tune VATT for vision and audio classification tasks and provide weights for the fine-tuned versions.

The backbone for vision is ViT, for audio WaveFormTransformer and for text they are using BERT/T5

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

Paper: https://arxiv.org/pdf/2104.11178.pdf
GitHub: https://github.com/google-research/google-research/tree/master/vatt

**Comment by fcakyon** [comment]
@johko have you started implementing it?

**Comment by fcakyon** [comment]
@johko I totally understand it. Interested in your implementation since I will be using VATT in my research next year :)

Are you working on a TF implementation?

**Comment by johko** [comment]
> @johko I totally understand it. Interested in your implementation since I will be using VATT in my research next year :)
> 
> Are you working on a TF implementation?

Sorry for the late reply (again 🙈). Yes I'm working on a TF implementation. As the original repo is using it, I'm first doing that and then see about pytorch.

**Comment by fcakyon** [comment]
@johko, thanks for the response! I may also help with the pytorch part once you finalize the TF implementation 👍

**Comment by johko** [comment]
@fcakyon that would be great, as my expertise is more in TF 🙂

**Comment by johko** [comment]
Hey @NielsRogge , I'm sorry but I think I have to stop working this for good. I'd love to finish it, but every time I think now I finally have some time to do it, something else comes around :disappointed: 

I think I just can't provide a big contribution like this atm and will rather focus on smaller things. But maybe @fcakyon wants to pick up on it.

Sorry for blocking this so long.

**Comment by pretbc** [comment]
any news about VATT PyTorch implementation ?

**Comment by NeighborhoodCoding** [comment]
Yap, there is VATT Pytorch implementation :
https://github.com/MattBortoletto/VATT-pytorch

**Comment by qgallouedec** [comment]
Hey @johko, sorry for the very late reply, we're catching up on the model-request backlog, and thanks for the effort you put into kicking this off (no worries at all about stepping back).

Since 2022 our process for new architectures has shifted to a *Model on the Hub first* approach: the recommended way to use VATT today is to publish the modeling code + weights in a Hub repo and load with `trust_remote_code=True`, no core port required. Given the original effort wound down a couple of years ago and no PR ever landed, I'm closing this as won't-implement for now. That's not a hard no, if you, @fcakyon, or anyone else wants to revive it, the cleanest route is a `modular_vatt.py` contribution following the modular guide (https://github.com/huggingface/transformers/blob/150eb7c9ed4091294c829fa0e9466b090cb0f87f/docs/source/en/modular_transformers.md) and the new-model-addition template
